algoliasearch-core/src/test/java/com/algolia/search/integration/client/ApiKeysTest.java

```diff
@@ -62,7 +63,17 @@ void testApiKeys() {
     DeleteApiKeyResponse deleteApiKey = searchClient.deleteApiKeyAsync(apiKey).join();
     deleteApiKey.waitTask();
 
-    RestoreApiKeyResponse restoreApiKey = searchClient.restoreApiKeyAsync(apiKey).join();
+    RestoreApiKeyResponse restoreApiKey;
+    while (true) {
+      try {
+        restoreApiKey = searchClient.restoreApiKeyAsync(apiKey).join();
+        break;
+      } catch (AlgoliaApiException e) {
+        if (e.getHttpErrorCode() == 404 && e.getMessage().equals("Key already exists")) {
```

Comment:
The engine doesn't commit on error messages :D But I guess that's fine for this test.

Reply:
@Ant-hem I'm not sure of what you mean here?

Reply:
@aseure He meant that the API engine may change this message "Key already exists" in the future. But is actually fine,  because we are using it in tests here.

Reply:
Exactly!